x/blob/README.md

```diff
@@ -2,44 +2,141 @@
-
-After a block has been created, the user can verify that their data was included in a block via a blob inclusion proof. A blob inclusion proof uses the `ShareCommitment` in the original transaction and subtree roots of the block's data square to prove to the user that the shares that compose their original data do in fact exist in a particular block.
+    1. `NamespaceIds []byte`: the namespaces they wish to publish each blob to.
+       The namespaces here must match the namespaces in the `Blob`s.
+    1. `ShareCommitment []byte`: a share commitment that is the root of a Merkle
+       tree where the leaves are share commitments to each blob associated with
+       this BlobTx.
+
+After the `BlobTx` is submitted to the network, a block producer separates the
+transaction from the blob. Both components get included in the data square in
+different namespaces: the BlobTx gets included in the PayForBlobNamespace (one
+of the [reserved
+namespaces](../../specs/src/specs/consensus.md#reserved-namespaces)) and the
+associated blob gets included in the namespace the user specified in the
+original `BlobTx`. Further reading: [Message Block
+Layout](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md)
+
+After a block has been created, the user can verify that their data was included
+in a block via a blob inclusion proof. A blob inclusion proof uses the
+`ShareCommitment` in the original transaction and subtree roots of the block's
+data square to prove to the user that the shares that compose their original
+data do in fact exist in a particular block.
 
 ## State
 
-The blob module doesn't maintain it's own state.
+The blob module doesn't maintain it's own state outside of a single param.
 
 When a `MsgPayForBlob` is processed, it consumes gas based on the blob size.
 
 ## Messages
 
-- [`MsgPayForBlob`](https://github.com/celestiaorg/celestia-app/blob/8b9c4c9d13fe0ccb6ea936cc26dee3f52b6f6129/proto/blob/tx.proto#L39-L44) pays for the blob to be included in the block.
-
-## PrepareProposal
+- [`MsgPayForBlobs`](https://github.com/celestiaorg/celestia-app/blob/v1.0.0-rc2/proto/celestia/blob/v1/tx.proto#L16-L31)
+  pays for a set of blobs to be included in the block. Transactions containing
+  this `sdk.Msg` are better known as "PFBs".
+
+```proto
+// MsgPayForBlobs pays for the inclusion of a blob in the block.
+message MsgPayForBlobs {
+  string signer = 1;
+  // namespaces is a list of namespaces that the blobs are associated with. A
+  // namespace is a byte slice of length 33 where the first byte is the
+  // namespaceVersion and the subsequent 32 bytes are the namespaceId.
```

Comment:
oof noticed that `33` is incorrect. Fixed in https://github.com/celestiaorg/celestia-app/pull/1951